### PR TITLE
Fix typings

### PR DESCRIPTION
**Pull request by benny-medflyt**
This is a more accurate model the parse results.

See also:
<https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pg-connection-string/index.d.ts>

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -2,13 +2,13 @@ export function parse(connectionString: string): ConnectionOptions;
...
-  password: string | null;
-  user: string | null;
-  port: number | null;
-  database: string | null;
-  client_encoding: string | null;
-  ssl: boolean | null;
+  password?: string;
+  user?: string;
+  port: string | null;
+  database: string | null | undefined;
```
Comment (hjr3) [inline index.d.ts]: An optional parameter essentially means `| undefined`. Should be `database?: string | null;`

File: index.d.ts [hunk 2]
```diff
@@ -2,13 +2,13 @@ export function parse(connectionString: string): ConnectionOptions;
...
+  client_encoding?: string | undefined;
```
Comment (hjr3) [inline index.d.ts]: The extra `| undefined` is redundant.

**Comment by benny-medflyt** [comment]
I've updated the pull request.

I've made a few more fields optional, here is an example where only `host` and `database` are returned:

```
> p.parse("/")
{ host: '/', database: undefined }
```

I also fixed the type of `ssl` (it can be a string).

I left the `database` field to not contain a `?` annotation, because there is a difference in TypeScript between a field being optional, vs a field that is required but may have the value "undefined"

**Comment by hjr3** [comment]
Thank you for following up @benny-medflyt .

**Pull request merged**